tests/unit/test_client.py

```diff
@@ -1732,6 +1741,76 @@ def test_get_signed_policy_v4_with_access_token(self):
         )
 
 
+@pytest.mark.parametrize("test_data", _POST_POLICY_TESTS)
+def test_conformance_post_policy(test_data):
+    import datetime
+    from google.cloud.storage.client import Client
+
+    in_data = test_data["policyInput"]
+    timestamp = datetime.datetime.strptime(in_data["timestamp"], "%Y-%m-%dT%H:%M:%SZ")
+
+    client = Client(credentials=_DUMMY_CREDENTIALS)
+
+    # mocking time functions
+    with mock.patch("google.cloud.storage._signing.NOW", return_value=timestamp):
+        with mock.patch(
+            "google.cloud.storage.client.get_expiration_seconds_v4",
+            return_value=in_data["expiration"],
+        ):
+            with mock.patch("google.cloud.storage.client._NOW", return_value=timestamp):
+
+                policy = client.generate_signed_post_policy_v4(
+                    bucket_name=in_data["bucket"],
+                    blob_name=in_data["object"],
+                    conditions=_prepare_conditions(in_data),
+                    fields=in_data.get("fields"),
+                    credentials=_DUMMY_CREDENTIALS,
+                    expiration=in_data["expiration"],
+                    virtual_hosted_style=in_data.get("urlStyle")
+                    == "VIRTUAL_HOSTED_STYLE",
+                    bucket_bound_hostname=in_data.get("bucketBoundHostname"),
+                    scheme=in_data.get("scheme"),
+                )
+    fields = policy["fields"]
+
+    for field in (
+        "x-goog-algorithm",
+        "x-goog-credential",
+        "x-goog-date",
+        "x-goog-signature",
+    ):
+        assert fields[field] == test_data["policyOutput"]["fields"][field]
+
+    out_data = test_data["policyOutput"]
+    decoded_policy = base64.b64decode(fields["policy"]).decode("unicode_escape")
+    assert decoded_policy == out_data["expectedDecodedPolicy"]
+    assert policy["url"] == out_data["url"]
+
+
+def _prepare_conditions(in_data):
+    """Helper for V4 POST policy generation conformance tests.
+
+    Сonvert conformance test data conditions dict into list.
+
+    Args:
+        in_data (dict): conditions arg from conformance test data.
+
+    Returns:
+        list: conditions arg to pass into generate_signed_post_policy_v4().
+    """
+    if "conditions" in in_data:
+        conditions = []
+        for key, value in in_data["conditions"].items():
+            # camel case to snake case with "-" separator
+            field = re.sub(r"(?<!^)(?=[A-Z])", "-", key).lower()
+
+            if isinstance(value, list):
```

Comment:
There are only two condition types in conformance tests. Could you instead use a string match otherwise this code block is mysterious.

Reply:
I've took a closer look, we actually can drop these statements at all, and keep such a code:
```python
for key, value in in_data["conditions"].items():
    # camel case to snake case with "-" separator
    field = re.sub(r"(?<!^)(?=[A-Z])", "-", key).lower()
    conditions.append([field] + value)

return conditions
```
There is only one value type in conformance tests. I've written code with `isinstance()` to make it more flexible in case of adding new conformance tests.

Pushing the new version with just `conditions.append([field] + value)`.